src/main/java/com/es/phoneshop/web/DeleteCartItemServlet.java

```diff
@@ -0,0 +1,38 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class DeleteCartItemServlet extends HttpServlet {
+
+    private CartService cartService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        cartService = DefaultCartService.getInstance();
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Long productId = parseProductID(request);
+        Cart cart = cartService.getCart(request);
+
+        cartService.delete(cart, productId);
+
+        response.sendRedirect(request.getContextPath() + "/cart?modalSuccess=Cart item deleted successfully");
+    }
+
+    private Long parseProductID(HttpServletRequest request) {
+        String productId = request.getPathInfo().substring(1);
+        return Long.valueOf(productId);
```

Comment:
We still need to add validation on back-end: for the valid number ID existing product.
Otherwise, the user can alter HTML code to pass the arbitrary product ID in a method call.